src/imageio/imageio_png.c

```diff
@@ -199,17 +199,39 @@ dt_imageio_retval_t dt_imageio_open_png(dt_image_t *img, const char *filename, d
     return DT_IMAGEIO_LOAD_FAILED;
   }
 
-  for(size_t j = 0; j < height; j++)
+  const size_t npixels = width * height;
```

Comment:
You should cast either width or height to size_t to get a size_t*size_t multiplication instead of a uint32_t multiplication that is only widened afterward, which would break for images with more than 4 gigapixels.  (PNG might be limited to 64K by 64K though, in which case it doesn't matter.)

Reply:
Thanks! Fixed.